### DISCUSSION TITLE
See-through, non-solid, light-blocking brush

### DISCUSSION THREAD

**Discussion by 4LT** [ericwa/ericw-tools#345] [Ideas]
Essentially a parallel to the Source engines's `tools/toolsblocklight` material.  These are useful to embed as thin brushes inside doors to prevent light bleeding through, but the door can't be made solid to light because that would create black areas in the door frame.  Idea had come up when talking to @Paril a while back where I mentioned I thought it was a bug that `skip` textured brush with `_shadow 1` doesn't block light.

**Comment by ericwa** [comment]
Yeah this sounds good. 

>  I mentioned I thought it was a bug that skip textured brush with _shadow 1 doesn't block light.

There is very limited support for this but it's implemented in a very hacky / inelegant way. The challenge is, `qbsp` discards `skip` faces entirely, so `light.exe` has nothing to raytrace against to generate shadows.

What I do is (this is old, has been in the release builds for several years):

 - if it's an actual bmodel like `func_wall` / `func_illusionary`,
 - and the entire bmodel is textured with `skip`,
 - and `_shadow 1` is set on the bmodel,

Then `light.exe` decompiles the bsp tree back into brushes, then creates faces, just for the purpose of being invisible shadow casters.


I think the better approach is for `qbsp` to recognize and keep these "invisible shadow casting" faces, either in an auxiliary file or bspx lump.  Then light can just use them directly without any fancy decompiling.

**Comment by 4LT** [comment]
Oh I see, it's not supported at a file format level, unless I make the brush an entity.  Is there much support for BSPX outside of FTE?  If not, could light decompose the BSPX to a BSP2?

**Comment by ericwa** [comment]
I think BSPX works everywhere (Dos/WinQuake included) and on all .bsp formats - it's just a standard way of stashing extra custom lumps at the end of a .bsp file and giving them names. Unknown BSPX lumps are just ignored by engines.

An auxiliary file to communicate between `qbsp` and `light` would work just as well for this, like the `.texinfo` file that is used for a lot of features currently. The only reason to put it in a BSPX lump would be, hypothetically, engines with dynamic shadows could read the lump and include the geometry when they cast dnyamic shadows.

**Reply by 4LT** [reply]
Ahh, wasn't aware of the texinfo file
